src/main/resources/org/broadinstitute/dsde/sam/liquibase/changesets/20201028_nested_roles.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="dummy"
+                   xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+
+    <changeSet logicalFilePath="dummy" author="mtalbott" id="nested_role_table">
+        <createTable tableName="SAM_NESTED_ROLE">
+            <column name="base_role_id" type="BIGINT">
+                <constraints nullable="false" foreignKeyName="FK_SNR_BASE_ROLE" referencedTableName="SAM_RESOURCE_ROLE" referencedColumnNames="id" primaryKey="true"/>
+            </column>
+            <column name="nested_role_id" type="BIGINT">
+                <constraints nullable="false" foreignKeyName="FK_SNR_NESTED_ROLE" referencedTableName="SAM_RESOURCE_ROLE" referencedColumnNames="id" primaryKey="true"/>
+            </column>
+            <column name="descendants_only" type="BOOLEAN" defaultValue="false">
+                <constraints nullable="false" primaryKey="true"/>
```

Comment:
all three columns have `primaryKey="true"`.  I would imagine that we want the `*_role_id` columns indexed, but probably not the `descendants_only` column.  Not sure if the primary key thing was a shorthand for getting the indexing or not, but either way, I don't think we'd want it on this column.

Reply:
@gpolumbo-broad This is how all the `descendants_only` columns are currently implemented. This is the only way that I'm aware of to guarantee that we can distinguish between `(2, 5, true)` and `(2, 5, false)`. If the primary key was just the `*_role_id` columns, it would cause a primary key violation as I understand it.

Reply:
oh, right, sure, that makes sense.